### ISSUE TITLE
Gabor's Data PyFixest Wishlist

### ISSUE THREAD

**Issue by s3alfisc** [py-econometrics/pyfixest#752]
I plan to tackle some of these over the holidays: 

- [ ] no prediction interval -> LOW EFFORT if no fixed effects supported, as PR prepared [by @greenguy33](https://github.com/py-econometrics/pyfixest/pull/573), more effort with fixed effects
- [ ] R2 for WLS -> LOW EFFORT [Issue](https://github.com/py-econometrics/pyfixest/issues/348)
- [ ] logit support -> HIGH EFFORT. [Issue: link](https://github.com/py-econometrics/pyfixest/pull/674)
- [ ] probit support -> HIGH EFFORT 
- [ ] HAC SEs -> MEDIAN EFFORT [Issue](https://github.com/py-econometrics/pyfixest/issues/675)
- [ ] locally defined functions cannot be used inside the formula (lspline) -> FEASIBILITY?
- [ ] ARIMA VAR -> WONT DO

**Comment by adamvig96** [comment]
Hi @s3alfisc,

I created a poc CL for context capture aka ` locally defined functions cannot be used inside the formula (lspline)` here: https://github.com/adamvig96/pyfixest/commit/96fa4fcf35fe03e7de67272a9b1e214e777ea0c8

I think it would be good to have this in pyfixest, apart from the fact that we'd need it in  the Gabor's DA book codebase – eg. statsmodels have this functionality. This is just a POC as I'm not familiar with the pyfixest codebase, but it worked for me in some of our case studies.

FYI @gbekes, we need this for logit and probit too as lsplines are used extensively there.

**Comment by s3alfisc** [comment]
Hi @adamvig96 , to make sure I understand the problem correctly - you need formulaic to parse a custom data transform (in this case a spline) and need to provide the extra context via the feols / fepois APIs? If yes I very much agree that this would be a very useful feature to have! 

If yes, I took a quick initial look at your code and I think it looks quite good already - I don't think there'd be much to add except maybe some tests (happy to help with that =) ).  If you open a PR, we can get started with getting this into pyfixest =)

**Comment by adamvig96** [comment]
Yes, I think that's an accurate description of the issue. I opened #769 and #770.

**Comment by s3alfisc** [comment]
Thanks!

**Comment by s3alfisc** [comment]
I will merge logit + probit tomorrow (I am encountering a strange bug that I currently cannot explain), will then continue with R2 for WLS. Then the prediction intervals, last HAC.

**Comment by s3alfisc** [comment]
We've just merged R2 for WLS (and within and adjusted R2 too)! [PR here](https://github.com/py-econometrics/pyfixest/pull/815), thanks to @AronNemeth. I am also finalizing the [prediction standard error PR ](https://github.com/py-econometrics/pyfixest/pull/573), I should finish it by tomorrow evening. Then only HAC will be left =)

**Comment by s3alfisc** [comment]
... and we've merged the prediction errors PR.

**Comment by DanielBarabas** [comment]
Hi, @s3alfisc, in [chapter 17](https://github.com/gabors-data-analysis/da_case_studies/blob/dev/ch17-predicting-firm-exit/ch17-predicting-firm-exit.ipynb) of the book we would like to show the AMEs of a logit. In the documentation you gave an example using the marginaleffects package that I tried to replicate (in the "Quick reminder about probability models" part of the notebook) but the avg_slopes function simply gives back the original logit coefficients, not the AMEs, as if it was an OLS regression. Have you encountered anything similar? Or do you have any ideas that we chould try?

On another note: in our conda environment  we could only use marginaleffects==0.0.11 because of a dependency issue regarding numpy (numba needs numpy<1.27, that comes with pyfixest, while marginaleffects=0.1.0 needs numpy>2.0.0), maybe this causes the issue?

**Comment by s3alfisc** [comment]
Hi @DanielBarabas , this can indeed be the case! 

To create the docs, I am using marginaleffects 0.0.14: 

```marginaleffects            0.0.14                                     147.1 KiB   pypi   marginaleffects-0.0.14-py3-none-any.whl
```
But I am also running higher numpy and numba versions? 
```
numba                      0.61.0                                     11.7 MiB    pypi   numba-0.61.0-cp312-cp312-win_amd64.whl
numpy                      2.1.3                                      40.2 MiB    pypi   numpy-2.1.3-cp312-cp312-win_amd64.whl
```
This is all under pyfixest 0.28.0 and the `docs` environment. I suppose you are running the same version as you are installing from pypi?

**Comment by DanielBarabas** [comment]
Thanks, I will check the numba bc that might be it (when i get to my laptop)

**Comment by DanielBarabas** [comment]
I managed to get the env working with the versions you specified, however the avg_slopes function now runs into the following error: 

![Image](https://github.com/user-attachments/assets/1165bc0f-d07b-4b65-81ef-9465308ae84c)

My hunch is that using numba automatically degrades numpy to <2, and will try to solve this (but as you might have noticed this is not my expertise)
